### PR TITLE
BLE integration

### PR DESCRIPTION
**Pull request by tianyouw**

### CODE REVIEW

File: app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java [hunk 1]
```diff
@@ -36,9 +36,11 @@ public BLEService(){
 
     }
 
+    public String deviceAddress;
+
     private final static String TAG = BLEService.class.getSimpleName();
 
-    private static final long SCAN_PERIOD = 10000;
+    private static final long SCAN_PERIOD = 30000;
```
Comment (atatmaja) [inline app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java:43]: what's the point of increasing the scan period here??

File: app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java [hunk 2]
```diff
@@ -341,19 +371,42 @@ public void onScanResult(int callbackType, ScanResult result) {
...
+
     //function to filter out force sensors
     public boolean isEspertoWatch(String name)
     {
         //TODO::update with new BLE chip
-        String watch_name = getString(R.string.esperto_device_name);
-        if(name.equals(watch_name)) return true;
-        else return false;
+        String watchName = getString(R.string.esperto_device_name);
+        return name.equals(watchName);
```
Comment (atatmaja) [inline app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java:409]: Do we really need this to be its own function, we could just do if(name.equals(getString(R.string.esperto_device_name)))

Comment (atatmaja) [reply app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java:409]: This is pretty minor though so it's fine to just not change if you don't feel like it, also I guess it sorta improves code readability

File: app/src/main/java/espertolabs/esperto_ble_watch/LoginActivity.java [hunk 3]
```diff
@@ -247,9 +229,17 @@ public void onClick(DialogInterface dialog, int id) {
...
+
+        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+        boolean isConnected = activeNetwork != null && activeNetwork.isConnectedOrConnecting();
+
+        if (isConnected) {
+            Intent registerUser = new Intent(v.getContext(), RegisterActivity.class);
+            startActivity(registerUser);
+            finish();
+        } else {
+            Toast.makeText(this, "Connect to WiFi or data then please try again :)", Toast.LENGTH_SHORT).show();
```
Comment (atatmaja) [inline app/src/main/java/espertolabs/esperto_ble_watch/LoginActivity.java]: Store this in strings.xml to be consistent with how we're handling error toasts elsewhere.

File: app/src/main/java/espertolabs/esperto_ble_watch/RegisterActivity.java [hunk 4]
```diff
@@ -59,7 +53,53 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
+        if (!manager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
+            buildAlertMessageNoGps();
+
+        }
+    }
+
+    private void buildAlertMessageNoGps() {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("Your location is disabled, do you want to enable it in order to search for an Esperto watch?")
```
Comment (atatmaja) [inline app/src/main/java/espertolabs/esperto_ble_watch/RegisterActivity.java]: strings.xml

File: app/src/main/java/espertolabs/esperto_ble_watch/RegisterActivity.java [hunk 5]
```diff
@@ -59,7 +53,53 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSION_REQUEST_COARSE_LOCATION: {
+                if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+
+                } else {
+                    Toast.makeText(this, "Location permissions are required for Bluetooth scanning", Toast.LENGTH_SHORT).show();
```
Comment (atatmaja) [inline app/src/main/java/espertolabs/esperto_ble_watch/RegisterActivity.java]: Strings.xml

File: app/src/main/java/espertolabs/esperto_ble_watch/ScanActivity.java [hunk 6]
```diff
@@ -105,21 +82,16 @@ protected void onCreate(Bundle savedInstanceState) {
 
         //Check whether BLE is supported
         if (!getPackageManager().hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE)) {
-            Toast.makeText(this, "Bluetooth LE is not supported.", Toast.LENGTH_SHORT).show();
+            Toast.makeText(this, "Bluetooth LE is not supported on this device", Toast.LENGTH_SHORT).show();
```
Comment (atatmaja) [inline app/src/main/java/espertolabs/esperto_ble_watch/ScanActivity.java]: Strings.xml

File: app/src/main/java/espertolabs/esperto_ble_watch/SummaryActivity.java [hunk 7]
```diff
@@ -225,13 +202,12 @@ public void onRequestPermissionsResult(int requestCode, String permissions[], in
                 if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
 
                 } else {
-                    Toast.makeText(this, "Location permissions are required for Bluetooth scanning.", Toast.LENGTH_SHORT).show();
+                    Toast.makeText(this, "Location permissions are required for Bluetooth scanning", Toast.LENGTH_SHORT).show();
```
Comment (atatmaja) [inline app/src/main/java/espertolabs/esperto_ble_watch/SummaryActivity.java]: strings.xml

File: app/src/main/java/espertolabs/esperto_ble_watch/SummaryActivity.java [hunk 8]
```diff
@@ -400,7 +376,9 @@ public void onServiceConnected(ComponentName className,
             mBLEService = binder.getService();
             mServiceBound = true;
             if (mBLEService.initialize()) {
+                mBLEService.scanFromSummary(true);
                 mBLEService.connect(user.getDeviceAddress());
+
```
Comment (atatmaja) [inline app/src/main/java/espertolabs/esperto_ble_watch/SummaryActivity.java]: minor but remove this extra line

**Comment by atatmaja** [comment]
No major issues, some pretty minor stuff to fix otherwise looks good to me.
Also don't forget to resolve merge conflicts!

File: app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java (hunk 1 above)
Comment (tianyouw) [reply app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java:43]: I was having some issues with it not detecting the watch, increased time to allow it to scan longer before quitting.

File: app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java (hunk 2 above)
Comment (tianyouw) [reply app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java:409]: Probably doesn't need to be its own method.

File: app/src/main/java/espertolabs/esperto_ble_watch/LoginActivity.java (hunk 3 above)
Comment (tianyouw) [reply app/src/main/java/espertolabs/esperto_ble_watch/LoginActivity.java]: Agreed.

File: app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java (hunk 2 above)
Comment (tianyouw) [reply app/src/main/java/espertolabs/esperto_ble_watch/BLEService.java:409]: Looks like its being used in tests. Keeping as is.

**Pull request merged**